src/hotspot/share/code/nmethod.cpp

```diff
@@ -1554,7 +1558,9 @@ oop nmethod::oop_at_phantom(int index) const {
 // notifies instanceKlasses that are reachable
 
 void nmethod::flush_dependencies(bool delete_immediately) {
-  DEBUG_ONLY(bool called_by_gc = Universe::heap()->is_gc_active() || Thread::current()->is_ConcurrentGC_thread();)
+  DEBUG_ONLY(bool called_by_gc = Universe::heap()->is_gc_active() ||
+                                 Thread::current()->is_ConcurrentGC_thread() ||
+                                 Thread::current()->is_Worker_thread();)
```

Comment:
Three places use the same condition. Did you consider creating a helper function?

Reply:
I wonder if adding a helper function would encourage people to think they're the same again.

Reply:
All three places doesn't quite have identical conditions today. We might want to clean this up, for example, I'd argue that at least one of the asserts could be removed. However, I'm not sure I want to do such a cleanup of a fairly unrelated thing as part of this PR.

Reply:
I agree.  I missed that one of the one of them is_at_safepoint and not is_gc_active.